src/mds/Locker.cc

```diff
@@ -890,7 +890,7 @@ void Locker::eval_gather(SimpleLock *lock, bool first, bool *pneed_issue, list<M
 	return;
       }
 
-      if (!mds->is_cluster_degraded() ||
+      if (!mds->is_cluster_degraded() &&
 	  mds->mdsmap->get_state(auth) >= MDSMap::STATE_REJOIN) {
```

Comment:
it's not wrong. !mds->is_cluster_degraded() check is an optimization, it avoids checking target mds' state for each message